xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-api/src/test/java/org/xwiki/panels/internal/AbstractPanelsUIExtensionManagerTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.panels.internal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Provider;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.component.util.DefaultParameterizedType;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.test.annotation.BeforeComponent;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectComponentManager;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+import org.xwiki.test.mockito.MockitoComponentManager;
+import org.xwiki.uiextension.UIExtension;
+import org.xwiki.uiextension.internal.WikiUIExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ComponentTest
+public class AbstractPanelsUIExtensionManagerTest
+{
+    @InjectComponentManager
+    private MockitoComponentManager componentManager;
+
+    @MockComponent
+    private ConfigurationSource configurationSource;
+
+    @MockComponent
+    @Named("currentmixed")
+    private DocumentReferenceResolver<String> resolver;
+
+    @InjectMockComponents
+    private LeftPanelsUIExtensionManager panelUIExtensionManager;
```

Comment:
Personally I'd have either:
1) created a TestablePanelsUIExtensionManager component in src/test/java and registered it as a component, since we're testing AbstractPanelsUIExtensionManager here

or 

2) Created a LeftPanelsUIExtensionManagerTest instead to test LeftPanelsUIExtensionManager and thus test AbstractPanelsUIExtensionManager at the same time